### PR TITLE
[SYSTEMML-1647] Verify whether StepLinearReg works with MLContext

### PR DESCRIPTION
**Pull request by iyounus**
StepLinearRegDS.dml script in SystemML cannot be used with new MLContext due to some write statements inside nested if statements. Also, there are some issues related to reordering of coefficients. This PR fixes these problems.

### CODE REVIEW

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1523/

**Comment by iyounus** [comment]
@prithvirajsen @ae2015 could you please review this. Thanks.

**Comment by iyounus** [comment]
@prithvirajsen Thank you for reviewing this. I've made changes as you suggested.

File: scripts/algorithms/StepLinearRegDS.dml [hunk 1]
```diff
@@ -25,20 +25,23 @@
...
+# S           String     ---   Location to write the selected features ordered as computed by the algorithm
+# O           String     " "   Location to write the printed statistics; by default is standard output
+# icpt        Int        0     Intercept presence, shifting and rescaling the columns of X:
+#                              0 = no intercept, no shifting, no rescaling;
+#                              1 = add intercept, but neither shift nor rescale X;
+#                              2 = add intercept, shift & rescale X columns to mean = 0, variance = 1
+# thr         Double    0.01   Threshold to stop the algorithm: if the decrease in the value of AIC falls below thr
+#                              no further features are being checked and the algorithm stops 
+# fmt         String   "text"  Matrix output format for B (the betas) only, usually "text" or "csv"
+# write_beta  Int       1      Should the beta's be returned?
```
Comment (deroneriksson) [inline scripts/algorithms/StepLinearRegDS.dml]: Thank you @iyounus. Could you update the docs here to boolean values?

File: scripts/algorithms/StepLinearRegDS.dml [hunk 2]
```diff
@@ -70,20 +73,22 @@
 #
 # HOW TO INVOKE THIS SCRIPT - EXAMPLE:
 # hadoop jar SystemML.jar -f StepLinearRegDS.dml -nvargs X=INPUT_DIR/X Y=INPUT_DIR/Y B=OUTPUT_DIR/betas
-#     O=OUTPUT_DIR/stats S=OUTPUT_DIR/selected icpt=2 thr=0.01 fmt=csv
+#     O=OUTPUT_DIR/stats S=OUTPUT_DIR/selected icpt=2 thr=0.01 fmt=csv write_beta=1
```
Comment (deroneriksson) [inline scripts/algorithms/StepLinearRegDS.dml]: Also update write_beta=TRUE

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1525/

File: scripts/algorithms/StepLinearRegDS.dml (hunk 2 above)
Comment (iyounus) [inline scripts/algorithms/StepLinearRegDS.dml]: done!

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1526/

**Comment by prithvirajsen** [comment]
Looks good. Thanks for making the changes.

**Comment by deroneriksson** [comment]
Thank you @iyounus. I'll merge.

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1598/

**Pull request closed without merging**